tests/solana_tests/simple.rs

```diff
@@ -280,3 +280,29 @@ fn two_arrays() {
 
     vm.constructor("two_arrays", &[], 0);
 }
+
+#[test]
+fn dead_storage_bug() {
+    let mut vm = build_solidity(
+        r#"
+        contract deadstorage {
+            uint public maxlen = 10000;
+            uint public z;
+            uint public v;
+
+            constructor() {
+                for(uint64 i = 0; i < 10; i++) {
+                    print("bitches0");
+                    uint x = i*10e34;
+                    print("bitches1");
+                    v = x%maxlen;
+                    print("bitches2");
+                    z =((0)%maxlen);
+                    print("bitches3");
```

Comment:
I see that dogs inspired you today 🐕‍🦺🦮🐶🐩

Reply:
ah that was not supposed to be in the PR 😳 

I've improved the test somewhat, now it tests more than just success.